x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/documents.tsx

```diff
@@ -9,35 +9,32 @@ import React from 'react';
 
 import { useValues } from 'kea';
 
-import { EuiPageHeader, EuiCallOut, EuiSpacer } from '@elastic/eui';
+import { EuiCallOut, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 
-import { FlashMessages } from '../../../shared/flash_messages';
-import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
-
 import { AppLogic } from '../../app_logic';
 import { EngineLogic, getEngineBreadcrumbs } from '../engine';
+import { AppSearchPageTemplate } from '../layout';
 
-import { DocumentCreationButton } from './components';
+import { DocumentCreationButton, EmptyState } from './components';
 import { DOCUMENTS_TITLE } from './constants';
 import { SearchExperience } from './search_experience';
 
 export const Documents: React.FC = () => {
-  const { isMetaEngine } = useValues(EngineLogic);
+  const { isMetaEngine, engine } = useValues(EngineLogic);
   const { myRole } = useValues(AppLogic);
 
   return (
-    <>
-      <SetPageChrome trail={getEngineBreadcrumbs([DOCUMENTS_TITLE])} />
-      <EuiPageHeader
-        pageTitle={DOCUMENTS_TITLE}
-        rightSideItems={
-          myRole.canManageEngineDocuments && !isMetaEngine
-            ? [<DocumentCreationButton />]
-            : undefined
-        }
-      />
-      <FlashMessages />
+    <AppSearchPageTemplate
+      pageChrome={getEngineBreadcrumbs([DOCUMENTS_TITLE])}
+      pageHeader={{
+        pageTitle: DOCUMENTS_TITLE,
+        rightSideItems:
+          myRole.canManageEngineDocuments && !isMetaEngine ? [<DocumentCreationButton />] : [],
+      }}
+      isEmptyState={!engine.document_count}
```

Comment:
Oh, totally forgot to mention - I have plans to pull this out to a flat `isEngineEmpty` selector, but I'm going to do that in a later/tech debt/polish PR, since I'll also be using that check for the engine overview